src/com/spikes2212/command/genericsubsystem/GenericSubsystem.java

```diff
@@ -0,0 +1,84 @@
+package com.spikes2212.command.genericsubsystem;
+
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+/**
+ * This class represents a generic subsystem that moves within a limitation, or
+ * without one.
+ *
+ * @author Omri "Riki" Cohen
+ */
+public abstract class GenericSubsystem extends Subsystem {
+
+	private double currentSpeed = 0;
+	private double maxSpeed;
```

Comment:
change `minSpeed` and `maxSpeed`  to be of type `Supplier<Double>`. with that, 3 constructors. 
* one which receives suppliers
* one which receives doubles
* a default constructor (boundries are -1, 1)